src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -160,71 +163,75 @@ class TezosNodeOperator(val node: TezosRPCInterface, batchConf: BatchFetchConfig
   }
 
   /**
-    * Gets all blocks from the head down to the oldest block not already in the database.
-    * @param network    Which Tezos network to go against
-    * @return           Blocks and Account hashes involved
+    * Given a level range, creates sub-ranges of max the given size
+    * @param levels a range of levels to partition into (possibly) smaller parts
+    * @param pageSize how big a part is allowed to be
+    * @return an iterator over the part, which are themselves `Ranges`
     */
-  def getBlocksNotInDatabase(network: String): Future[List[(Block, List[AccountId])]] =
-    for {
-      maxLevel <- ApiOperations.fetchMaxLevel
-      blockHead <- getBlockHead(network)
-      headLevel = blockHead.metadata.header.level
-      headHash = blockHead.metadata.hash
-      blocks <-
-        if (headLevel <= maxLevel) {
-          logger.info("No new blocks to fetch from the network")
-          Future.successful(List.empty)
-        } else {
-          if (maxLevel == -1) logger.warn("There were apparently no blocks in the database. Downloading the whole chain..")
-          else logger.info("Found new block head at level {}, currently stored max is {}. Fetching missing blocks", headLevel, maxLevel)
-          getBlocks(network, headHash, maxLevel + 1, headLevel)
-        }
-    } yield blocks
+  def partitionBlocksRanges(levels: Range.Inclusive, pageSize: Int = 1000): Iterator[Range.Inclusive] =
```

Comment:
Please ensure `pageSize` is read from the Lorre conf file.

Reply:
done